hummingbot/connector/gateway_EVM_AMM.py

```diff
@@ -911,9 +905,22 @@ async def update_balances(self, on_interval=False):
             self._last_balance_poll_timestamp = current_tick
             local_asset_names = set(self._account_balances.keys())
             remote_asset_names = set()
+
+            # get
+            config: Optional[Dict[str, str]] = GatewayConnectionSetting.get_connector_spec_from_market_name(self._name)
+            other_tokens: Set[str] = set()
+            if config is not None:
+                other_tokens: Set[str] = set(config.get("tokens", "").split(","))
+                other_tokens.discard("")
+
+            other_tokens.update(self._tokens)
+            other_tokens.update([self._native_currency])
+            tokens_to_query: List[str] = list(other_tokens)
+
             resp_json: Dict[str, Any] = await GatewayHttpClient.get_instance().get_balances(
-                self.chain, self.network, self.address, list(self._tokens) + [self._native_currency]
+                self.chain, self.network, self.address, tokens_to_query
```

Comment:
Similar to previous comment, we shouldn't handle this here if we don't want to keep doing thesame in every other generic connectors.
We can handle this functionality in the `connect_market` function in `UserBalances` class.